app/Services/VersionService.php

```diff
@@ -138,12 +142,43 @@ public function createOrUpdateVersion(string $versionString, bool $allowed) : bo
      */
     public function toggleVersionAllowed(string $versionString)
     {
-        $version = Version::where('version', '=', $versionString);
+        $version = Version::withTrashed()->where('version', '=', $versionString)->first();
```

Comment:
Could you do firstOrFail here and just catch ModelNotFoundException and then re-throw your custom exception?

Reply:
You could do to achieve the same end - I like the predictability of return null and handle, personally :) Exceptions are... fun!